.github/workflows/sonarcloud.yml

```diff
@@ -8,6 +8,7 @@ on:
 
 jobs:
   build:
+    if: github.repository_owner == 'damianszczepanik'
```

Comment:
1. leave comments in the code why is that
2. with this change there is no need to have branch limitation in line 5-6

Reply:
Added variation of the first comment above 'if'.  Adjusted the original.  I think you do still want it only running against master, that is unless you are looking at pull requests.  I can only speak for repos I work on and all simply run on master and nothing else.  If you are looking to more broadly look at those then sure, will remove that.

Reply:
In general - agree with you. My experience is that sonar quite frequently has problem with authentication because of token that is lost somehow. Quite often I did not notice the problem with sonar token and for a long time sonar was not working. Having this on PR (even for myself ) gives opportunity to find such issues earlier :)
Best would be to process sonar against every PR to see tech debt earlier but I was not able to do that :(

Reply:
Also this allows me to run sonar on fix-branch before fix goes to master so I can perform the validation and confirm the fix before it goes to master :)

Reply:
@damianszczepanik Per https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/, unless you are a paying customer ($160 a year), its not possible to run on pull requests.  So I've adjusted this a bit to indicate that and point to the documentation.